analysis_notebooks/Attribute Analysis.ipynb

```diff
@@ -696,46 +898,56 @@
-    "        if first_pass:\n",
-    "            to_write[3] = ['(att_cnt) Distribution of object categories each attribute appears with, sorted by ratio between the max count attribute and min count attribute.']\n",
-    "            plt.savefig(\"results/{0}/{1}/1.png\".format(folder_name, save_loc))\n",
-    "        plt.show()\n",
+    "        if ordinal_plot:\n",
+    "            p_trend, cat_ord_options = [], []\n",
+    "            for idx, count in enumerate(np.array(supercategory_counts).T):\n",
+    "                cat_trend = np.array(count)[axis]\n",
+    "                _, _, _, p, _ = stats.linregress([x for x in range(len(axis))], cat_trend)\n",
+    "                p_trend.append(p)\n",
+    "                cat_ord_options.append((f\"{datasets.DEFAULT_GROUPINGS_TO_NAMES[idx + 1]}, p={dec_to_show(p)}\", idx))\n",
+    "            cat_ord_options = [x for _, x in sorted(zip(p_trend, cat_ord_options))]    \n",
+    "            return cat_ord_options, np.array(supercategory_counts)[axis].T\n",
+    "        else:\n",
+    "            fig = plt.figure(figsize=(10, 6))\n",
+    "            r1 = np.arange(len(datasets.DEFAULT_GROUPINGS_TO_NAMES)-1)\n",
+    "            r1 = [r1[i]+i*num_attrs*barWidth for i in range(len(r1))]\n",
+    "            r = [r1]\n",
+    "            for i in range(1, num_attrs+1):\n",
+    "                r.append([x + barWidth for x in r[len(r)-1]])\n",
+    "\n",
+    "            category_attr_ratios = [0 for i in range(len(supercategory_counts[0]))]\n",
+    "            max_categories = [0 for i in range(len(supercategory_counts[0]))]\n",
+    "            for category in range(len(supercategory_counts[0])):\n",
+    "                max_category = 0\n",
+    "                min_category = 1\n",
+    "                max_attr = -1\n",
+    "                for attr in range(num_attrs):\n",
+    "                    if supercategory_counts[attr][category] > max_category:\n",
+    "                        max_category = supercategory_counts[attr][category]\n",
+    "                        max_attr = attr\n",
+    "                    if supercategory_counts[attr][category] < min_category:\n",
+    "                        min_category = supercategory_counts[attr][category]\n",
+    "                category_attr_ratios[category] = max_category / min_category\n",
+    "                max_categories[category] = max_attr\n",
+    "\n",
+    "            order = np.argsort(category_attr_ratios)\n",
+    "            r = np.array(r)\n",
+    "            for i in range(num_attrs-1, -1, -1):\n",
+    "                plt.barh(r[i], supercategory_counts[i][order], height=barWidth, color=COLORS[i], edgecolor='white', label=attr_names[i])\n",
+    "\n",
+    "            ticks = [r[0]+(num_attrs/2)*barWidth for i in range(len(r1))][0]\n",
+    "            plt.yticks(ticks, np.array([datasets.DEFAULT_GROUPINGS_TO_NAMES[i+1] for i in range(len(r1))])[order], fontsize=fontsize)\n",
```

Comment:
instead of datasets.DEFAULT_GROUPINGS_TO_NAMES you should use dataset.supercategories_to_names (the latter is set to the former in the case of COCO, but that is not always going to be the case). this change applies to all instances of datasets.DEFAULT_GROUPINGS_TO_NAMES